src/main/java/com/redhat/digkins/util/DiggerClientException.java

```diff
@@ -0,0 +1,30 @@
+package com.redhat.digkins.util;
+
+import java.util.TreeMap;
+
+/**
+ * Represents internal client exception
+ */
+public class DiggerClientException extends Exception {
+
+  public DiggerClientException() {
+    super();
+  }
+
+  public DiggerClientException(String message) {
+    super(message);
+  }
+
+  public DiggerClientException(String message, Throwable cause) {
+    super(message, cause);
+  }
+
+  public DiggerClientException(Throwable cause) {
+    super(cause);
+  }
+
+  @Override
+  public Throwable fillInStackTrace() {
```

Comment:
What we doing here is called lightweight exception and is typical pattern for applications who choose to use exceptions to direct application flow. Normally java apps return exceptions if something bad is happening and it would require some stack trace check later. In some cases developers may chose exceptions as way to define application flow. For example: `CustomerAlreadyRegisteredException`. In our case library we using is using that pattern so I'm wrapping their exceptions into our for better error handling. That exception may appear during the normal application execution. Statement above is used to not pollute logs with stack trace because it has absolutely no actual meaning here. I have no strong feeling to have it.

TL;DR; That's typical pattern for Client based apps when you using Exceptions as way to communicate with users code.